### PR TITLE
Fatresize

### PR DESCRIPTION
**Pull request by ssgen**
Add the fatresize package to nixpkgs
(http://sourceforge.net/projects/fatresize/)

### CODE REVIEW

File: pkgs/tools/filesystems/fatresize/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  '';
+  
+  makeFlags = ''
+    LDFLAGS=-lparted-fs-resize
+  '';
+  
+  propagatedBuildInputs = [ parted utillinux ];
+  
+  meta = {
+    description = "The FAT16/FAT32 non-destructive resizer.";
```
Comment (nckx) [inline pkgs/tools/filesystems/fatresize/default.nix]: The full stop is unnecessary. See http://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -15251,5 +15251,7 @@ let
   udevil = callPackage ../applications/misc/udevil {};
 
   facter = callPackage ../tools/system/facter {};
+  
+  fatresize = callPackage ../tools/filesystems/fatresize {};
```
Comment (nckx) [inline pkgs/top-level/all-packages.nix]: `all-packages.nix` is certainly a bit of a mess, but please move this to the `### TOOLS` section (respecting the more-or-less alphabetical order as much as possible).

File: pkgs/tools/filesystems/fatresize/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, parted, substituteAll, utillinux, ... }:
```
Comment (nckx) [inline pkgs/tools/filesystems/fatresize/default.nix]: The `...` can be dropped (any extra parameters can't be accessed within the expression anyway).

File: pkgs/tools/filesystems/fatresize/parted_nix.patch [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+diff --git a/configure b/configure
+index ed31457..a7012b7 100755
+--- a/configure
++++ b/configure
+@@ -4456,7 +4456,7 @@ if test "${ac_cv_lib_parted_ped_device_read+set}" = set; then
+   echo $ECHO_N "(cached) $ECHO_C" >&6
+ else
+   ac_check_lib_save_LIBS=$LIBS
+-LIBS="-lparted  $LIBS"
++LIBS="-lparted $LIBS"
```
Comment (nckx) [inline pkgs/tools/filesystems/fatresize/parted_nix.patch]: Is this really needed to fix the build? (I know nothing about autocruft.)

File: pkgs/tools/filesystems/fatresize/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    "unpackPhase"
+    "patchPhase"
+    "configurePhase"
+    "buildPhase"
+    "installPhase"
+  ];
+  
+  buildInputs = [ parted utillinux ];
+  
+  patches = [ ./parted_nix.patch ];
```
Comment (nckx) [inline pkgs/tools/filesystems/fatresize/default.nix]: This doesn't seem to add anything Nix-specific like the file name would imply. Could you elaborate a bit more (preferably in the patch header) where this patch comes from, what exactly it does and why?

File: pkgs/tools/filesystems/fatresize/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, parted, substituteAll, utillinux, ... }:
+
+stdenv.mkDerivation rec {
+  
+  version = "1.0.2";
+  name = "fatresize-${version}";
+
+  src = fetchurl {
+    url = mirror://sourceforge/fatresize/fatresize-1.0.2.tar.bz2;
```
Comment (nckx) [inline pkgs/tools/filesystems/fatresize/default.nix]: You can use `"...${name}.tar.bz2"` here unless you have reason not to (update scripts or somesuch).

**Comment by nckx** [comment]
Please squash both commits into one following the scheme from `CONTRIBUTING.md`.

File: pkgs/tools/filesystems/fatresize/default.nix [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    sha256 = "04wp48jpdvq4nn0dgbw5za07g842rnxlh9mig4mslz70zqs0izjm";
+  };
+  
+  phases = [
```
Comment (jagajaga) [inline pkgs/tools/filesystems/fatresize/default.nix]: Remove `phases` section. As I can see you have `preConfigure` not mentioned here.

**Comment by jagajaga** [comment]
And please rename commits' name as it said in `CONTRIBUTING.md`.

**Comment by nckx** [comment]
@jagajaga: I'm sure they'll read the feedback when they have time...

File: pkgs/tools/filesystems/fatresize/default.nix (hunk 1 above)
Comment (ssgen) [inline pkgs/tools/filesystems/fatresize/default.nix]: Hi,

I have read your mails - sorry for the late reply ...
I am starting to correct the issues you've pointed out :)
Thanks for your support

2015-08-18 1:02 GMT+03:00 Tobias Geerinckx-Rice <notifications@github.com>:

File: pkgs/tools/filesystems/fatresize/parted_nix.patch (hunk 4 above)
Comment (ssgen) [inline pkgs/tools/filesystems/fatresize/parted_nix.patch]: No, not really - I like things aligned though :)))
Reverting to original :P

2015-08-18 1:10 GMT+03:00 Tobias Geerinckx-Rice <notifications@github.com>:

**Comment by ssgen** [comment]
Hello,

Reading right now - will modify as necessary
Thanks for the pointer :)

2015-08-18 1:14 GMT+03:00 Tobias Geerinckx-Rice <notifications@github.com>:

File: pkgs/tools/filesystems/fatresize/default.nix (hunk 7 above)
Comment (ssgen) [inline pkgs/tools/filesystems/fatresize/default.nix]: Removing ...

2015-08-19 14:44 GMT+03:00 Arseniy Seroka <notifications@github.com>:

**Comment by ssgen** [comment]
I think it should be good now ...
Regards

**Comment by jagajaga** [comment]
Please rebase.

**Comment by jagajaga** [comment]
Leave just one commit please :)

**Comment by ssgen** [comment]
Done ...
I have a question though:
How do I get rid of:
error: failed to push some refs to 'https://github.com/ssgen/nixpkgs.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details. 
without pushing with -f ?

**Comment by jagajaga** [comment]
You can't push without `-f` because you need to rewrite.

**Pull request merged**